enhancements/sig-policy/98-long-term-compliance-history/README.md

```diff
@@ -0,0 +1,455 @@
+process the Kubernetes event and add the status entry in the local and hub replicated policy that owns the evaluated
+policy (i.e. in the `policy-templates` array). A policy's history is limited to 10 events.
+
+##### Recording a Compliance Event
+
+As previously mentioned, the compliance events will be stored in a relational database. This database will not be
+managed by Open Cluster Management and is limited to just Postgresql at the momement. The connection information for
+this database will be in a secret called `governance-policy-database` in the `open-cluster-management` namespace and
+require a key of `connectionURL` in the format of `postgres://username:password@localhost:5432/database_name`. For
+security reasons, each managed cluster cannot directly write to the database. Instead, an HTTP endpoint is to be added
+to the Policy Propagator to perform authorization checks and translating compliance events to the appropriate SQL
+commands. This is preferred so that the database design doesn't require a separate database (not server) per cluster for
+least privilege authorization. Note that this HTTP endpoint will only start if the `governance-policy-database` secret
+exists. Technically speaking, a separate goroutine would watch for the creation of this secret.
+
+The authorization code would leverage the Go APIs in
+[github.com/stolostron/rbac-api-utils](https://github.com/stolostron/rbac-api-utils/blob/159deac7d398c3470ae730c3fbb8ffbbcca1ac5a/pkg/rbac/rbac.go#L187)
+or [kube-rbac-proxy](https://pkg.go.dev/github.com/brancz/kube-rbac-proxy@v0.14.2/pkg/authz#NewSarAuthorizer), and
+require the Kubernetes service account recording the compliance event to have write access to the `status` subresource
+of the `policies.policy.open-cluster-management.io` resource in the applicable managed cluster namespace on the hub. The
+Status Sync controller already has this access. Below is an example of a role on the hub that could record compliance
+events for a managed cluster called `cluster1`.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: policy-status-writer
+  namespace: cluster1
+rules:
+  - apiGroups:
+      - policy.open-cluster-management.io
+    resources:
+      - policies/status
+    verbs:
+      - patch
+```
+
+The HTTP endpoint would be at `/api/v1/compliance-events` and would accept a `POST` request with the following schema:
+
+```json
+{
+  "cluster": { "name": "cluster1" },
+  "parent_policy": { "name": "policies.etcd-encryption" },
+  "policy": {
+    "apiGroup": "policy.open-cluster-management.io",
+    "kind": "ConfigurationPolicy",
+    // Optional if specHash is provided and an entry already exists for that spec hash in the database.
+    // This approach saves on network bandwidth since the spec can be quite large.
+    "spec": {},
+    // Optional if spec is provided
+    "specHash": {}
```

Comment:
Would it just always provide the `specHash`?  I expect the "spec" is only initialized on the hub by the propagator so managed clusters don't get to define a new spec.

Reply:
That's a good idea. The propagator *could* create entries for `spec` in the `policies` table so only the `specHash` needs to be provided from the managed cluster. That does simplify things. I'll update the doc with that.